crates/test/src/lib.rs

```diff
@@ -74,6 +74,49 @@ pub mod __macro_support {
 
     const RECORD_STORAGE: *const c_char = sys::c_string!("storage");
 
+    pub fn parse_args<'a>(args: *mut u8) -> Args<'a> {
+        if args.is_null() {
+            Args("")
+        } else {
+            let args_cstr = unsafe { CStr::from_ptr(args.cast_const().cast()) };
```

Comment:
ATM the function is unsound since an invalid pointer can be passed to it (it is not `unsafe`) and yet the `args` has a number of [restrictions](https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#safety) on it.

It should either be validated or the function should be marked `unsafe`.

Reply:
IIRC my thinking was that this is validated because we're paying in a pointer from the Flipper Zero that we know that (or rather, by definition) the pointer is to a valid C string. However, it's true that because I moved this into a macro helper, it could be used by others. So yes, I'll make this `unsafe`.